runtime/sema/errors.go

```diff
@@ -1819,10 +1819,9 @@ func (*UnsupportedOverloadingError) isSemanticError() {}
 func (*UnsupportedOverloadingError) IsUserError() {}
 
 func (e *UnsupportedOverloadingError) Error() string {
-	return fmt.Sprintf(
-		"%s overloading is not supported yet",
-		e.DeclarationKind.Name(),
-	)
+	var initializerName string = "init"
+	return fmt.Sprint(
+		"cannot redeclare initializer `%s` is already declared", initializerName)
```

Comment:
Good idea to rephrase the error message from a missing feature to simply what is wrong. 

However, the error is not just used for initializers, but for various declarations (`DeclarationKind`). Please generalize it again, by using `e.DeclarationKind.Name()`

Reply:
Maybe instead of rephrasing the message, switch to using `RedeclarationError` in places where `UnsupportedOverloadingError` is used, and remove `UnsupportedOverloadingError` altogether.

Reply:
@turbolent @SupunS thank you for review I i will update it